activesupport/lib/active_support/encrypted_configuration.rb

```diff
@@ -38,5 +45,10 @@ def serialize(config)
       def deserialize(config)
         config.present? ? YAML.load(config) : {}
       end
+
+      # Validate that +contents+ is correct format as YAML
+      def validate(contents) # :doc:
+        YAML.load(contents, content_path)
```

Comment:
Ah, is it because passing `content_path` ensures people can see that what file aborted the save?

Reply:
Yes, it is. There are differences in error messages as shown below.

```
# with `content_path` 
.rbenv/versions/2.4.1/lib/ruby/2.4.0/psych.rb:377:in `parse': (/home/yaginuma/program/rails/master/config/credentials.yml.enc): did not find expected alphabetic or numeric character while scanning an alias at line 1 column 8 (Psych::SyntaxError)

# without `content_path`
.rbenv/versions/2.4.1/lib/ruby/2.4.0/psych.rb:377:in `parse': (<unknown>): did not find expected alphabetic or numeric character while scanning an alias at line 1 column 8 (Psych::SyntaxError)
```